rpcs3/Emu/RSX/GL/GLGSRender.cpp

```diff
@@ -1681,6 +1681,23 @@ void GLGSRender::flip(int buffer, bool emu_flip)
 			image = m_flip_tex_color->id();
 		}
 
+		if (m_frame->screenshot_toggle == true)
+		{
+			m_frame->screenshot_toggle = false;
+
+			std::vector<u8> sshot_frame(buffer_height * buffer_width * 4);
+
+			if (gl::get_driver_caps().ARB_dsa_supported)
+				glGetTextureImage(image, 0, GL_BGRA, GL_UNSIGNED_INT_8_8_8_8, buffer_height * buffer_width * 4, sshot_frame.data());
+			else
+				glGetTextureImageEXT(GL_TEXTURE_2D, GL_TEXTURE_2D, 0, GL_BGRA, GL_UNSIGNED_INT_8_8_8_8, sshot_frame.data());
```

Comment:
No, argument 0 is the texture name which is the GLuint handle generated by OpenGL runtime. It should be `image` in this case.

Reply:
Two recommendations:
1. You need to create a gl::pixel_pack_settings struct  and bind it to reset the pixel pack settings. There is no guarantee that the last user of the pixel download pipeline reset the options to default.
2. You're grabbing BGA with 8_8_8_8 encoding, which is writing a byteswapped u32 (ARGB) to memory. Use GL_UNSIGNED_BYTE as the type to get your data as actual BGRA which will match vulkan and allow you to use a TRANSFORM_BGR to store your data cleanly as RGBA. You can then eliminate the separate path in the gs screenshot method and have cleaner code.